test/lib/ansible_test/_internal/commands/sanity/mypy.py

```diff
@@ -0,0 +1,247 @@
+"""Sanity test which executes mypy."""
+from __future__ import annotations
+
+import dataclasses
+import os
+import re
+import typing as t
+
+from . import (
+    SanityMultipleVersion,
+    SanityMessage,
+    SanityFailure,
+    SanitySuccess,
+    SanitySkipped,
+    SanityTargets,
+    create_sanity_virtualenv,
+)
+
+from ...constants import (
+    CONTROLLER_PYTHON_VERSIONS,
+    REMOTE_ONLY_PYTHON_VERSIONS,
+)
+
+from ...test import (
+    TestResult,
+)
+
+from ...target import (
+    TestTarget,
+)
+
+from ...util import (
+    SubprocessError,
+    display,
+    parse_to_list_of_dict,
+    ANSIBLE_TEST_CONTROLLER_ROOT,
+    ApplicationError,
+    is_subdir,
+)
+
+from ...util_common import (
+    intercept_python,
+)
+
+from ...ansible_util import (
+    ansible_environment,
+)
+
+from ...config import (
+    SanityConfig,
+)
+
+from ...host_configs import (
+    PythonConfig,
+    VirtualPythonConfig,
+)
+
+
+class MypyTest(SanityMultipleVersion):
+    """Sanity test which executes mypy."""
+    ansible_only = True
+
+    def filter_targets(self, targets):  # type: (t.List[TestTarget]) -> t.List[TestTarget]
+        """Return the given list of test targets, filtered to include only those relevant for the test."""
+        return [target for target in targets if os.path.splitext(target.path)[1] == '.py' and (
+                target.path.startswith('lib/ansible/') or target.path.startswith('test/lib/ansible_test/_internal/'))]
+
+    @property
+    def error_code(self):  # type: () -> t.Optional[str]
```

Comment:
Looks like you mix type comments with type annotations (at the very bottom) in this module — why not use a single style?

Reply:
Nearly all of the type annotations in `ansible-test` are comment based. The main exception is for dataclasses, which require native type annotations.